src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/Methodologies/MethodologyRepositoryTests.cs

```diff
@@ -148,6 +158,51 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
             MockUtils.VerifyAllMocks(methodologyParentRepository);
         }
 
+        [Fact]
+        public async Task GetLatestByPublication_MethodologyHasNoVersions()
+        {
+            var publication = new Publication();
+
+            var methodology = new MethodologyParent
+            {
+                Versions = new List<Methodology>()
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Publications.AddAsync(publication);
+                await contentDbContext.MethodologyParents.AddAsync(methodology);
+                await contentDbContext.PublicationMethodologies.AddRangeAsync(
+                    new PublicationMethodology
+                    {
+                        Publication = publication,
+                        MethodologyParent = methodology,
+                        Owner = true
+                    });
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var methodologyParentRepository = new Mock<IMethodologyParentRepository>(MockBehavior.Strict);
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.Attach(methodology);
```

Comment:
Ooh, what's this in aid of?

Reply:
The test uses a mock for the `MethodologyParentRepository`.
The `methodology` (parent) is returned by that `MethodologyParentRepository` mock when the service calls `_methodologyParentRepository.GetByPublication(publication.Id)`

The service then goes on to lazily load `methodology.Versions` from the database.

So this test sets up the versions so they can be loaded, but the database needs to be aware of `methodology` otherwise you get an error like 

> Navigation property 'Versions' on entity of type 'MethodologyParent' cannot be loaded because the entity is not being tracked.

Reply:
cool nice one!